File: Daily Exercise/旋转链表.java
public class ListNode {
    int val;
    ListNode next;
    ListNode() {}
    ListNode(int val) { this.val = val; }
    ListNode(int val, ListNode next) { this.val = val; this.next = next; }
}

class Solution {
    public ListNode rotateRight(ListNode head, int k) {
        if (head == null || k == 0) {
            return head;
        }

        int lenght = 0;
        while(head != null) {
            head = head.next;
            lenght++;
        }

        int s = k % lenght;

        ListNode slow = head;
        ListNode fast = head;

        for(int i =0; i < s; i++) {
            fast = fast.next;
        }

        while(fast.next != null) {
            slow = slow.next;
            fast = fast.next;
        }

        fast.next = head;
        head = slow.next;
        slow.next = null;

        return head;
    }
}

File: Daily Exercise/旋转链表.py
class Solution(object):
    def rotateRight(self, head, k):
        if k == 0 or not head or not head.next:
            return head

        l = 1
        tail = head
        while tail.next:
            l += 1
            tail = tail.next

        pos = l - k % l

        if pos == 1:
            return head
        
        curNode = head
        for _ in range(pos - 1):
            curNode = curNode.next
        

        tempHead = curNode.next
        curNode.next = None
        tail.next = head

        return tempHead

